### PR TITLE
WIP: Add missing field types

### PR DESCRIPTION
**Pull request by mficzel**
- `Neos.Fusion.Form:Input`:
- `Neos.Fusion.Form:Hidden`:
- `Neos.Fusion.Form:Password`
- `Neos.Fusion.Form:Radio`
- `Neos.Fusion.Form:Checkbox`
- `Neos.Fusion.Form:Checkbox.Multiple`
- `Neos.Fusion.Form:Upload`
- `Neos.Fusion.Form:Button`

### CODE REVIEW

File: Resources/Private/Fusion/Prototypes/Radio.fusion [hunk 1]
```diff
@@ -0,0 +1,13 @@
+prototype(Neos.Fusion.Form:Radio)  < prototype(Neos.Fusion.Form:Field) {
+    attributes = Neos.Fusion:DataStructure
+    value = ''
+    renderer = afx`
+        <input
+            {...props.attributes}
+            type="radio"
+            name={field.name}
+            value={props.value}
+            checked={field.value == this.value}
```
Comment (skurfuerst) [inline Resources/Private/Fusion/Prototypes/Radio.fusion]: Hey,

if I understand the code correctly, you need to set "field.value" for e.g. textboxes etc; but for radio boxes you need to set "value" directly.

Correct? If so, this wouldn't be good API-wise -- and I'd say having an external API which is stable and makes sense is crucial.

File: Resources/Private/Fusion/Prototypes/Checkbox.Mutiple.fusion [hunk 2]
```diff
@@ -0,0 +1,13 @@
+prototype(Neos.Fusion.Form:Checkbox.Multiple)  < prototype(Neos.Fusion.Form:Field) {
+    attributes = Neos.Fusion:DataStructure
+    value = ''
+    renderer = afx`
+        <input
+            {...props.attributes}
+            type="checkbox"
+            name={field.name}
+            value={props.value}
+            checked={(field.value && (Array.indexOf(field.value, this.value) > -1)) ? true : false}
```
Comment (skurfuerst) [inline Resources/Private/Fusion/Prototypes/Checkbox.Mutiple.fusion]: Nice change :) 

However I think for the "average integrator" it's hard to grasp the difference between props.value and field.value; as you need to specify both...

Comment (skurfuerst) [reply Resources/Private/Fusion/Prototypes/Checkbox.Mutiple.fusion]: Aaaah. I think if we use the eel Syntax you proposed with

```
prototype(Neos.Fusion.Form:Field)  < prototype(Neos.Fusion:Component) {
    @context.field = ${Form.fieldDefinition(form, this.name, this.property, this.value)}
    name = null
    property = null
    value = null
}

prototype(Neos.Fusion.Form:Form)  < prototype(Neos.Fusion:Component) {
    @context.form = ${Form.formDefinition(this.request, this.name, this.object, this.fieldNamePrefix)}
    request = ${request}
    name =  null
    object = null
    fieldNamePrefix = null
}
```

We won't have this problem, and he can just set "value" and that's it. Right?

Comment (mficzel) [reply Resources/Private/Fusion/Prototypes/Checkbox.Mutiple.fusion]: Yes i think getting rid of field.value will make it way easier to grasp

**Comment by mficzel** [comment]
@skurfuerst i merge this to get a complete set of fieldtypes. I also git rid of the `this`- references other than during creation of `field` and `form`. 

I am now quite satisfied with the afx-api those fields offer now (mostly `name`, `property`, `value` and `attributes`). Everything else is not meant to be final.

**Comment by skurfuerst** [comment]
:+1: <3

**Pull request merged**